data-products/src/snowflake_query_extraction/snowflake_query_extraction_flow.py

```diff
@@ -0,0 +1,244 @@
+    extraction_sql_template = """copy into @{stage_name}/{sql_name}/{year}/{month}/{day}/{hour}/{unix}/data
+    from ({sql})
+    header = true
+    overwrite = true
+    max_file_size = 104857600"""  # noqa: E501
+
+    # stage name must be one of our external GCS stages
+    stage_name = get_gcs_stage("gcs_pocket_shared")
+
+    # leverage UTC datetime for file partitioning
+    ts = now(tz="utc")
+
+    # get the query to use as extraction source
+    script_path = get_script_path()
+    sql_text_template = Path(
+        f"{script_path}/sql/{sf_extraction_input.sql_name}/data.sql"
+    ).read_text()
+    # render the query using inputs
+    sql_text = sql_text_template.format(
+        offset_key=sf_extraction_input.offset_key,
+        current_offset=current_offset,
+        new_offset=new_offset,
+        **sf_extraction_input.kwargs,
+    )
+    # return the extract job for downstream usage
+    return SfExtractionJob(
+        extraction_sql=extraction_sql_template.format(
+            stage_name=stage_name,
+            sql_name=sf_extraction_input.sql_name,
+            sql=sql_text,
+            year=ts.year,
+            month=ts.month,
+            day=ts.day,
+            hour=ts.hour,
+            unix=ts.int_timestamp,
+        ),
+        persist_state_sql=PERSIST_STATE_SQL.format(
+            sql_name=sf_extraction_input.sql_name, new_offset=new_offset
+        ),
+    )
+
+
+@flow(
+    description="""Main workflow for orchestration of query based extractions
+from Snowflake.
+"""
+)
+async def main(sf_extraction_input: SfExtractionInputs):
+    # get standard Prefect logger for logging
+    logger = get_run_logger()
+    # get reusable snowflake connector block for pocket
+    sfc = get_pocket_snowflake_connector_block()
+    # get the current offset using input model property
+    current_offset = await snowflake_query.with_options(  # type: ignore
+        name="get-current-offset"
+    )(
+        query=sf_extraction_input.current_offset_sql,
+        snowflake_connector=sfc,
+    )
+    logger.info(f"Current offset state result: {current_offset[0][0]}...")
+    # get the new offset using input model property
+    new_offset = await snowflake_query.with_options(  # type: ignore
+        name="get-new-offset"
+    )(
+        query=sf_extraction_input.new_offset_sql,
+        snowflake_connector=sfc,
+    )
+    logger.info(f"New offset will be: {new_offset[0][0]}...")
+    # get the extraction job details
+    extraction_job = create_extraction_job(
+        sf_extraction_input, current_offset[0][0], new_offset[0][0]
+    )
+    logger.info("Running extraction...")
+    # run the extraction
+    extract = await snowflake_query.with_options(name="run-extraction")(  # type: ignore
+        query=extraction_job.extraction_sql,
+        snowflake_connector=sfc,
+    )
+    logger.info("Applying new state...")
+    # commit the new offset
+    await snowflake_query.with_options(name="persist-new-offset")(  # type: ignore
```

Comment:
I dont remember do we need to return anything to confirm success or is it assumed if there are no exceptions?